pkg/datapath/linux/node_addressing.go

```diff
@@ -59,6 +50,28 @@ func listLocalAddresses(family int) ([]net.IP, error) {
 	return addresses, nil
 }
 
+func filterLocalAddresses(addrs []netlink.Addr, ipsToExclude []net.IP, addrScopeMax int) []net.IP {
+	var filteredIPs []net.IP
+	for _, addr := range addrs {
+		// This address is at a scope which is more narrow (numerically greater than) the configured
+		// max address scope. For example, if this addr is SCOPE_NOWHERE, and our addrScopeMax is
+		// SCOPE_LINK, then we do NOT treat the address as a local address. Similarly, if this addr
+		// is SCOPE_HOST, and our addrScopeMax is SCOPE_LINK, we do NOT treat the address as a local
+		// address.
+		if addr.Scope > addrScopeMax {
```

Comment:
@borkmann Should this be "<"? Are we trying to exclude "more universal" addresses with this "max" or are we trying to exclude "more narrow" addresses?

Reply:
I suppose it's obvious from 080857bdedca that it's intended to be the "most narrowly scoped" address.